docs/source/userdoc/requirements/interns/emina-mesic/ClockInit_requirements/non-functional_requirements.rst

```diff
@@ -0,0 +1,45 @@
+==================================================================
+Non-Functional Requirements for the Clock Initialization Function
+==================================================================
+.. req:: label
+   :id: REQ_INT3_1732228089
+   :status: Draft
+   :date-released:
+   :priority: Low
+   :submitted-by: Emina Mesic
+   :modified-by:
+   :category: Non-Functional
+   :safety-asil: 
+   :references: 
+   :verification-and-validation: 
+
+   The non-functional requirements address performance, scalability, and reliability of the clock initialization function. They define constraints for response times, error handling, precision, and resource usage, ensuring the system operates efficiently and robustly. 
+
+
+
+Performance and Response Time:
+  1. The system must initialize the clock configuration within 2 milliseconds under typical operating conditions.
+  2. The stabilization waiting period for the HSE (HSERDY bit) must not exceed 1 millisecond.
+  3. The stabilization waiting period for the PLL (PLLRDY bit) must not exceed 1 millisecond.
+
+Scalability:
+  4. The clock initialization function must support reconfiguration for clock frequencies between 16 MHz to 100 MHz without requiring code changes, provided valid PLLM, PLLN, and PLLP values are supplied.
+  5. The function must accommodate additional clock sources (e.g., MSI, HSI) with minimal adjustments, ensuring modularity.
```

Comment:
minimal adjustments nam ne govori puno, potrebno je definisati koji su to adjustments